### ISSUE TITLE
Video Backdrop and Theme song playback

### ISSUE THREAD

**Issue by Thursty24** [jellyfin/jellyfin-androidtv#1150]
Hi guys

Great job on Jellyfin!

Just one request for now:
Please add support for Video Backdrop and Theme song playback when browsing the library, just like the web interface

**Comment by Kaschdn** [comment]
I would also love to have background theme songs in the library as I have them on the web-client or the normal android client.

**Comment by Hackintosys** [comment]
+1

**Comment by KruyKaze** [comment]
+1

**Comment by SysCrasher** [comment]
In both cases enable the user specific settings in your client and rescan after placing the files.

Theme songs: 
Install the "Theme Songs"-Plugin (https://github.com/danieladov/jellyfin-plugin-themesongs not sure if it's needed to play theme songs by default). If you need your own theme song, place it as theme.mp3 in the root directory of your specific movie.

Theme video:
In the root directory of your movie create folder "backdrops" and move your own video file into it.

**Comment by Kaschdn** [comment]
> In both cases enable the user specific settings in your client and rescan after placing the files.
> 
> Theme songs: Install the "Theme Songs"-Plugin (https://github.com/danieladov/jellyfin-plugin-themesongs not sure if it's needed to play theme songs by default). If you need your own theme song, place it as theme.mp3 in the root directory of your specific movie.
> 
> Theme video: In the root directory of your movie create folder "backdrops" and move your own video file into it.

Are you sure they work for the Android-**TV-Client**?

**Comment by SysCrasher** [comment]
I can't confirm it for Android-TV, but theme songs work with the Android app on my Phone and Tablet.
For TV you can use an additional pc, if it doesn't work native.

**Comment by Kaschdn** [comment]
I know it's working for the other clients but **not** for the Android TV - client which is why @Thursty24 and me were asking for it. Using a PC for the TV is not an option. 

> I can't confirm it for Android-TV, but theme songs work with the Android app on my Phone and Tablet. For TV you can use an additional pc, if it doesn't work native.

**Comment by darinmanica** [comment]
+1 from me.  Theme music for TV series and movies creates a polished user experience when browsing a collection.

**Comment by nielsvanvelzen** [comment]
This is something I'd like to add and should be possible when #1057 is ready.

**Comment by ReenigneArcher** [comment]
+1, I'm the developer of [Themerr](https://github.com/LizardByte/Themerr-jellyfin) and a user just reported that it doesn't work on Android TV.

**Comment by Starfoxfs** [comment]
Yes doesn´t work on Android TV, Nvidia Shield TV Pro 2019.  Player Version 0.16.04

Would be very nice to have this Feature.

**Comment by Djfazer** [comment]
+1 not working on my two android tv:s and would be great if would :)
Ps. Theme songs working on my android phone and tizen tv.

**Comment by Yeetov** [comment]
> I would also love to have background theme songs in the library as I have them on the web-client or the normal android client.

Same, is it too much of a hassle to set up?

**Comment by deanaba** [comment]
This is a minor request in the grand scheme of things, but in comparing the web and mobile client to AndroidTV there's a lot of envy for those of us who mostly watch on a full size TV. Since show themes are working in the android client, perhaps it's not a big lift to get it working on AndroidTV?

**Comment by Chuacheee** [comment]
Hi, is there any news on this? I have had several customers for months who can listen to the BSO in the background in the library but still can not listen to it on androidtv.

**Comment by seelenfusion01** [comment]
Would also love when sounds played in Android TV and in the Android app <3
Thanks a lot for this masterpiece of Software

**Comment by Webierta** [comment]
I join this request, thank you

**Comment by mehrmoudi** [comment]
> This is something I'd like to add and should be possible when https://github.com/jellyfin/jellyfin-androidtv/issues/1057 is ready.

Hey @nielsvanvelzen, I'd like to contribute to this issue.

I'm wondering if this is actually blocked by #1057, or the legacy media player can still be used to at least play the theme music? If the legacy player can work, I'll look into implementing this using the interface you added in https://github.com/jellyfin/jellyfin-androidtv/pull/2453.

**Comment by nielsvanvelzen** [comment]
If this feature is implemented it **must** use the new playback code, this may be doable for theme songs but video support is not ready yet so video backdrops won't work. It should also NOT be done with the MediaManager as that is also meant to be removed eventually.

**Comment by mehrmoudi** [comment]
Gotcha. I personally care more about the theme songs, so I'd be happy to start just working on that. Video backdrop support can be added separately in the future.

For tracking, we should probably split this issue into 2 separate ones, one of music and one video. I can start working on the music one later today. If you could share a few code pointers that'd be great, if not I'll figure it out.

**Comment by nielsvanvelzen** [comment]
Doing only theme songs for now sounds like a good approach yeah. We can just leave this issue open and update the title once that is merged.

If you have any development related questions you can always pop into our Android development chat on Matrix/Discord. It is listed at jellyfin.org/contact.

**Comment by mehrmoudi** [comment]
https://github.com/jellyfin/jellyfin-androidtv/pull/4653 is ready for review.

**Comment by akhilmulpurii** [comment]
I’ve opened a PR that adds theme song playback support for movies and TV shows: [#5099](https://github.com/jellyfin/jellyfin-androidtv/pull/5099)￼.

I’m primarily a web developer and have just started learning Kotlin and Android architecture for the sake of implementing this feature, hence it might not be fully aligned with Android best practices — but it works well functionally. Feedback and reviews would be really helpful to refine it further!

**Comment by mattsigal** [comment]
Any chance this PR could be incorporated? I really miss the theme songs when switching from other devices to my TV.

**Comment by abhcoide** [comment]
> Any chance this PR could be incorporated? I really miss the theme songs when switching from other devices to my TV.

I don't think this PR will be incorporated in its present state.   It is a nice feature, which I think adds to the overall browsing of TV shows.

**Comment by ppwaskie** [comment]
> > Any chance this PR could be incorporated? I really miss the theme songs when switching from other devices to my TV.
> 
> I don't think this PR will be incorporated in its present state. It is a nice feature, which I think adds to the overall browsing of TV shows.

Is there a reason the PR won't be merged in the present state?  I read the PR and it looks like there was some minor discussion, and then it died on the vine.

I'm a C developer (kernel dev) so I have no business going near Kotlin or anything with a UI involved.  But I'd really like to see this audio feature make it across the line somehow.  I've been trying to switch the family to Jellyfin from Plex after 12 years of running the latter, and the little things with feature parity make it a harder sell.
